src/Cache/Dragon/Dragon.js

```diff
@@ -14,51 +14,30 @@ export default function CacheDragonCost() {
       Game.dragonLevel < 25 &&
       Game.dragonLevels[Game.dragonLevel].buy.toString().includes('sacrifice')
     ) {
-      let target = Game.dragonLevels[Game.dragonLevel].buy.toString().match(/Objects\[(.*)\]/)[1];
+      let target = "";
       const amount = Game.dragonLevels[Game.dragonLevel].buy
         .toString()
         .match(/sacrifice\((.*?)\)/)[1];
-      if (target !== 'i') {
-        target = target.replaceAll("'", '');
+      let cost = 0;
+      CopyData();
+      Object.keys(Game.Objects).forEach((j) => {
+        target = j;
         if (Game.Objects[target].amount < amount) {
           CacheCostDragonUpgrade = 'Not enough buildings to sell';
-        } else {
-          let cost = 0;
-          CopyData();
-          for (let i = 0; i < amount; i++) {
-            let price =
-              SimObjects[target].basePrice *
-              Game.priceIncrease **
-                Math.max(0, SimObjects[target].amount - 1 - SimObjects[target].free);
-            price = Game.modifyBuildingPrice(SimObjects[target], price);
-            price = Math.ceil(price);
-            cost += price;
-            SimObjects[target].amount -= 1;
-          }
-          CacheCostDragonUpgrade = `Cost to rebuy: ${Beautify(cost)}`;
+          return;
         }
-      } else {
-        let cost = 0;
-        CopyData();
-        Object.keys(Game.Objects).forEach((j) => {
-          target = j;
-          if (Game.Objects[target].amount < amount) {
-            CacheCostDragonUpgrade = 'Not enough buildings to sell';
-            return;
-          }
-          for (let i = 0; i < amount; i++) {
-            let price =
-              SimObjects[target].basePrice *
-              Game.priceIncrease **
-                Math.max(0, SimObjects[target].amount - 1 - SimObjects[target].free);
-            price = Game.modifyBuildingPrice(SimObjects[target], price);
-            price = Math.ceil(price);
-            cost += price;
-            SimObjects[target].amount -= 1;
-          }
-          CacheCostDragonUpgrade = `Cost to rebuy: ${Beautify(cost)}`;
-        });
-      }
+        for (let i = 0; i < amount; i++) {
```

Comment:
Sorry to intrude but why don't we use this function ([BuyBuilding](https://github.com/CookieMonsterTeam/CookieMonster/blob/195d7335f8be7d7f259a4c8cec48eea3abf4e92c/src/Sim/SimulationEvents/BuyBuilding.js)) instead of calculating it here?
Something like 
```js
BuildingGetPrice(
    target,
    SimObjects[target].basePrice,
    SimObjects[target].amount - amount,
    SimObjects[target].free,
    amount)
```